src/figment.rs

```diff
@@ -410,6 +522,15 @@ impl Figment {
     ///     assert!(not_a_dict.extract_inner::<bool>("cat").is_err());
     ///     assert!(not_a_dict.extract_inner::<bool>("dog").is_err());
     ///
+    ///     let cat_lives = root.focus("lives.0");
+    ///     assert_eq!(cat_lives.extract_inner::<u8>("cat").unwrap(), 9);
+    ///     let dog_lives = root.focus("lives.1");
+    ///     assert_eq!(dog_lives.extract_inner::<u8>("dog").unwrap(), 1);
```

Comment:
I wanted to demonstrate that key => index => key was possible, though I suppose I can simplify it.

Reply:
The additional array is actually crucial in demonstrating that focus works on arrays, since `focus("cat")` and `focus("cat.0")` both don't work.

```rust
figment::Jail::expect_with(|jail| {
    jail.create_file("Config.toml", r#"
        cat = [1, 2, 3]
        dog = [4, 5, 6]
    "#)?;

    let root = Figment::from(Toml::file("Config.toml"));
    assert_eq!(root.extract_inner::<u8>("cat.0").unwrap(), 1);
    assert_eq!(root.extract_inner::<u8>("dog.2").unwrap(), 6);

    let cat = root.focus("cat");
    // This should work if `root.extract("cat.1")` were equal to
    // `root.focus("cat").extract("1")` but this property only holds true
    // for dictionaries.
    assert_eq!(cat.extract_inner::<u8>("1").unwrap(), 2);

    Ok(())
});
```

So focusing _through_ an array onto a nested dictionary (`lives.0`) works, but focusing directly _on_ an array (`lives`) or a  non-dict in an array (`cat.0`) does not.
I think this is a reasonable restriction, but should be properly documented.

Thoughts?